index.js

```diff
@@ -40,6 +40,22 @@ sheets["sheets"].forEach(s => {
                         if (sheetResponse.statusCode == 200) {
                             var file = fse.createWriteStream(path.join(folder, rowFileName(data, headers) + ".csv"));
                             sheetResponse.pipe(file);
+                        } else if (sheetResponse.statusCode == 307) {
```

Comment:
Can there be multiple redirections (loop)? Is there any js http client to handle these implicitly?

Reply:
Yes, I have updated it. I checked the NodeJS documentation, but found that the native `https` module does not handle request redirection implicitly and we need to handle it manually. However, at the same time I found another js module called `follow-redirects` (https://github.com/follow-redirects/follow-redirects), which can help us handle the request forward. In my tests, I found that I could handle the redirection problem without changing any of the original code structure by introducing just three additional lines of code.